client/influxdb.go

```diff
@@ -178,19 +197,43 @@ func (c *Client) Query(q Query) (*Response, error) {
 	defer resp.Body.Close()
 
 	var response Response
-	dec := json.NewDecoder(resp.Body)
-	dec.UseNumber()
-	decErr := dec.Decode(&response)
+	if q.Chunked {
+		cr := NewChunkedResponse(resp.Body)
+		for {
+			r, err := cr.NextResponse()
+			if err != nil {
+				// Ignore EOF errors if we got an invalid status code.
+				if err == io.EOF && resp.StatusCode != http.StatusOK {
```

Comment:
I think that got in there because an earlier version I worked on did have it return EOF. The reason for keeping it there was because EOF can be returned if the server sends a 500. It doesn't appear like it matters where in the code this check is, so I'm going to leave it in the chunked response and remove it from here.

Good catch.